crates/table/src/table.rs

```diff
@@ -562,41 +564,65 @@ impl Table {
         self.blob_store_bytes -= blob_bytes;
     }
 
-    /// Insert row identified by `ptr` into indices.
+    /// Insert row identified by `new` into indices.
     /// This also checks unique constraints.
     /// Deletes the row if there were any violations.
     ///
-    /// SAFETY: `self.is_row_present(row)` must hold.
+    /// If `CHECK_SAME_ROW`, upon a unique constraint violation,
+    /// this will check if it's a duplicate row.
```

Comment:
What does this do if `CHECK_SAME_ROW` is false upon a unique constraint violation?